tests/ui/derives/lack-of-perfect-derives.stderr

```diff
@@ -0,0 +1,240 @@
+help: consider annotating `X` with `#[derive(Clone)]`
+   |
+LL + #[derive(Clone)]
+LL | struct X;
+   |
+
+error[E0599]: the method `eq` exists for struct `S<X, X, X>`, but its trait bounds were not satisfied
+  --> $DIR/lack-of-perfect-derives.rs:32:7
+   |
+LL | struct S<T, K, V>(Arc<T>, Rc<K>, Arc<V>, ());
+   | ----------------- method `eq` not found for this struct because it doesn't satisfy `S<X, X, X>: Iterator` or `S<X, X, X>: PartialEq`
+...
+LL | struct X;
+   | -------- doesn't satisfy `X: PartialEq`
+...
+LL |     s.eq(s2);
+   |       ^^ method cannot be called on `S<X, X, X>` due to unsatisfied trait bounds
+   |
+note: trait bound `X: PartialEq` was not satisfied
+  --> $DIR/lack-of-perfect-derives.rs:4:17
+   |
+LL | #[derive(Clone, PartialEq)]
+   |                 ^^^^^^^^^ unsatisfied trait bound introduced in this `derive` macro
+LL | struct S<T, K, V>(Arc<T>, Rc<K>, Arc<V>, ());
+   |          -  -  - implicit unsatisfied bound on this type parameter
+   |          |  |
+   |          |  implicit unsatisfied bound on this type parameter
+   |          implicit unsatisfied bound on this type parameter
+   = note: `#[derive(PartialEq)]` introduces `where`-bounds requiring every type parameter to implement `PartialEq`, even when not strictly necessary
+   = help: you can manually implement `PartialEq` with more targeted bounds: `impl<T, K, V> PartialEq for S<T, K, V> where Arc<T>: PartialEq, Rc<K>: PartialEq, Arc<V>: PartialEq { /* .. */ }`
+   = note: the following trait bounds were not satisfied:
+           `S<X, X, X>: Iterator`
+           which is required by `&mut S<X, X, X>: Iterator`
+note: the trait `Iterator` must be implemented
+  --> $SRC_DIR/core/src/iter/traits/iterator.rs:LL:COL
+help: consider annotating `X` with `#[derive(PartialEq)]`
+   |
+LL + #[derive(PartialEq)]
+LL | struct X;
+   |
+
+error[E0599]: the method `clone` exists for struct `N<X, Y, X>`, but its trait bounds were not satisfied
+  --> $DIR/lack-of-perfect-derives.rs:34:16
+   |
+LL | struct N<T, K, V>(Arc<T>, T, Rc<K>, K, Arc<V>, V, ());
+   | ----------------- method `clone` not found for this struct because it doesn't satisfy `N<X, Y, X>: Clone`
+...
+LL | struct X;
+   | -------- doesn't satisfy `X: Clone`
+...
+LL |     let n2 = n.clone();
+   |                ^^^^^ method cannot be called on `N<X, Y, X>` due to unsatisfied trait bounds
+   |
+note: trait bound `X: Clone` was not satisfied
+  --> $DIR/lack-of-perfect-derives.rs:8:10
+   |
+LL | #[derive(Clone, PartialEq)]
+   |          ^^^^^ unsatisfied trait bound introduced in this `derive` macro
+LL | struct N<T, K, V>(Arc<T>, T, Rc<K>, K, Arc<V>, V, ());
+   |          -     - implicit unsatisfied bound on this type parameter
+   |          |
+   |          implicit unsatisfied bound on this type parameter
+   = note: `#[derive(Clone)]` introduces `where`-bounds requiring every type parameter to implement `Clone`
```

Comment:
Note the mixed use of `Arc<T>` and `T`. We don't suggest `where Arc<T>: Clone, T: Clone`, so in the end we suggest no impl.